# fix duplicate symbols in sim/ppc sources
# - declare 'extern' to avoid allocation of the symbols for each file that includes sim/ppc/ld-insn.h
# - allocate those variables in sim/ppc/ld-insn.c only
# 
--- sim/ppc/ld-insn.h.orig	2014-06-11 16:34:42.000000000 +0000
+++ sim/ppc/ld-insn.h	2023-09-12 13:47:17.677392900 +0000
@@ -200,25 +200,25 @@
  table_include *includes,
  cache_table **cache_rules);
 
-model *models;
-model *last_model;
+extern model *models;
+extern model *last_model;
 
-insn *model_macros;
-insn *last_model_macro;
+extern insn *model_macros;
+extern insn *last_model_macro;
 
-insn *model_functions;
-insn *last_model_function;
+extern insn *model_functions;
+extern insn *last_model_function;
 
-insn *model_internal;
-insn *last_model_internal;
+extern insn *model_internal;
+extern insn *last_model_internal;
 
-insn *model_static;
-insn *last_model_static;
+extern insn *model_static;
+extern insn *last_model_static;
 
-insn *model_data;
-insn *last_model_data;
+extern insn *model_data;
+extern insn *last_model_data;
 
-int max_model_fields_len;
+extern int max_model_fields_len;
 
 extern void insn_table_insert_insn
 (insn_table *table,
--- sim/ppc/ld-insn.c.orig	2014-06-11 16:34:42.000000000 +0000
+++ sim/ppc/ld-insn.c	2023-09-12 14:04:28.924210287 +0000
@@ -28,6 +28,26 @@
 
 #include "igen.h"
 
+model *models;
+model *last_model;
+
+insn *model_macros;
+insn *last_model_macro;
+
+insn *model_functions;
+insn *last_model_function;
+
+insn *model_internal;
+insn *last_model_internal;
+
+insn *model_static;
+insn *last_model_static;
+
+insn *model_data;
+insn *last_model_data;
+
+int max_model_fields_len;
+
 static void
 update_depth(insn_table *entry,
 	     lf *file,
